Further Improve Incident Reporting (Fall Update, V2) (#2)
ryancdickson/www.ccadb.org: cas/incident-report.md

- help community members understand the relevant architectures, implementations, operations, and external dependencies surrounding an incident;
- demonstrate a detailed understanding of the root cause(s) of an incident;
- unambiguously explain how systems, processes, and/or policies failed; 
- describe the circumstances that prevented the problem(s) from being detected earlier;
- describe a clear timeline of a CA Owner's actions while responding to and remediating an incident;
- include a detailed and measurable explanation of actions taken or planned by the CA Owner that demonstrate a substantive commitment of how the CA Owner's systems, policies, and/or processes will be made more robust (i.e., demonstrate continuous improvement); and
- share lessons learned that could be helpful to all CA Owners in building better systems policies, and/or processes.

#### What does Root Cause Analysis consider?

Effective Root Cause Analysis (RCA) minimally considers the following points:

> 1. **Focus on the "why," not just the "what"**. RCAs shouldn't just identify what went wrong but should delve deeper into why it happened. This often involves going beyond the immediate technical or policy failure and considers contributing factors like human error, process deficiencies, or system design flaws.

Reviewer: ```suggestion
1. **Focus on the "why," not just the "what"**. RCAs go beyond identifying what went wrong and delve deeper into why it happened. This often involves looking past the immediate technical or policy failure and considers contributing factors like human error, process deficiencies, or system design flaws.
```
Removing "should" language.

2. **Use a systematic approach**. Employ structured methodologies like the following approaches to help ensure a thorough and organized investigation:
   - [**Chesterton’s Fence**](https://fs.blog/chestertons-fence/): Before changing anything, understand why it's there. This helps avoid unintended consequences and ensures solutions address the root cause, not just symptoms.

>    - [**5 Whys**](https://en.wikipedia.org/wiki/Five_whys): Repeatedly ask "why" to uncover the underlying cause of a problem. This simple technique helps drill down to the core issue and prevents superficial solutions.

Reviewer: ```suggestion
   - [**5 Whys**](https://en.wikipedia.org/wiki/Five_whys): Repeatedly ask "why" as many times as necessary to uncover the underlying cause of a problem. This simple technique helps drill down to the core issue and prevents superficial solutions.
```
To emphasize that it can be more or less than 5 times.

   - [**CATWOE**](https://www.toolshero.com/problem-solving/catwoe-analysis/): Analyze a problem from different perspectives (customers, actors, transformation, world view, owner, environmental constraints). This ensures a holistic understanding of the issue and its context.
   - [**Cause and Effect / Fishbone or Ishikawa**](https://en.wikipedia.org/wiki/Ishikawa_diagram): Visually map potential causes of a problem, categorizing them to identify root causes and relationships. This promotes systematic exploration and prevents overlooking factors.
   - [**Drilling Down**](https://sigma.software/about/media/problem-solving-techniques-part-two#2.-drill-down): Break complex problems into smaller, more manageable parts. This allows for focused investigation and helps identify specific areas for improvement.
   - **SRE Handbook Guidance**: These chapters provide practical advice on managing incidents effectively, including communication, coordination, and postmortem analysis. They emphasize a blameless culture focused on learning and improvement.
       - [Chapter 14](https://sre.google/sre-book/managing-incidents/)
       - [Chapter 15](https://sre.google/sre-book/postmortem-culture/)

> 4. **Consider all potential contributing factors**. RCAs should consider a broad range of potential causes, including technical issues, policy issues, human factors, process breakdowns, and external influences. It's crucial to avoid jumping to conclusions or focusing solely on the most obvious cause(s).

Reviewer: ```suggestion
4. **Consider all potential contributing factors**. RCAs consider a broad range of potential causes, including technical issues, policy issues, human factors, process breakdowns, and external influences. It's crucial to avoid jumping to conclusions or focusing solely on the most obvious cause(s).
```

5. **Collect and analyze data**. Data is critical for supporting RCA conclusions. This might involve reviewing logs, monitoring metrics, internal incident reports, and other relevant information to identify patterns and anomalies.

> 6. **Involve relevant stakeholders**. RCAs should be a collaborative effort involving engineers, operators, support teams, and other relevant stakeholders. This helps ensure a diverse range of perspectives and expertise is considered.

Reviewer: ```suggestion
6. **Involve relevant stakeholders**. RCAs are a collaborative effort involving engineers, operators, support teams, and other relevant stakeholders. This helps ensure a diverse range of perspectives and expertise is considered.
```

7. **Prioritize action items**. The ultimate goal of an RCA is to prevent future incidents caused by the same failures. Therefore, it's important to identify and prioritize actionable recommendations that address the root causes identified.

> 8. **Focus on blameless postmortems**. Emphasize a blameless culture when conducting postmortems. The focus should be on learning from the incident and improving systems, not on assigning blame or punishing individuals.

Reviewer: ```suggestion
8. **Focus on blameless postmortems**. Emphasize a blameless culture when conducting postmortems. The focus is on learning from the incident and improving systems, not on assigning blame or punishing individuals.
```

> 9. **Continuously improve the process**. RCA is an iterative process. Organizations should continuously refine their RCA approaches based on lessons learned from previous incidents and evolving best practices.

Reviewer: ```suggestion
9. **Continuously improve the process**. RCA is an iterative process where an organization continuously refines its approach based on lessons learned from previous incidents and evolving best practices.
```

- Sharing lessons learned from past experiences.

Individuals representing CA Owners are especially encouraged to participate broadly in the reporting processes, extending their contributions beyond incidents involving only their own organization. Sharing insights and perspectives across organizational boundaries fosters a collaborative learning environment and strengthens the overall security posture of the Web PKI ecosystem.

To ensure productive dialogue, please keep all comments constructive, relevant to the report, and in line with the [CCADB Code of Conduct](https://docs.google.com/document/d/19ALqEvHtTE6OUTz2FaOXrU9gruIdvia5EDh3hXeGpZA/edit#heading=h.cumc0pgd1s7c).

#### What Bugzilla account can I use?

**For individuals affiliated with a Publicly-Trusted CA Owner:**
- **To better encourage blamelessness**, when posting incident reports or responding to comments on reports for which they are affiliated, participants are encouraged to respond from a Bugzilla account associated with one of the CA e-mail aliases disclosed to the CCADB, rather than an individual contributor’s account.

> - **To better respect a desire for individual privacy and potential risk of retaliation**, individuals participating in the reporting process should feel welcome to participate **responsibly** from an account that does not identify the individual posting or their organizational affiliation.

Reviewer: ```suggestion
- **To better respect a desire for individual privacy and potential risk of retaliation**, individuals participating in the reporting process are welcome to participate **responsibly** from an account that does not identify the individual posting or their organizational affiliation.
```

| **Timeline summary*** | Describe (1) when the non-compliance began (2), when the non-compliance was detected, and (3) when the non-compliance ended or is expected to end. |
| **Relevant policies** | Describe the policy name(s), applicable version(s), and corresponding section(s) that result in this problem being diagnosed as an incident. |
| **Source of incident disclosure*** | Choice of "Self Reported", "Third Party Reported", or "Audit". |

*Note*: If notified of an incident by an external, third party reporter, please respect their privacy by *only* disclosing their name if affirmatively approved to do so (e.g., say "we received a report from a community member" instead of explicitly naming individuals). Fields marked with an asterisk (*) are not required in Preliminary Incident Reports if submitted by a third party reporter.

**Impact:** The Impact section contains a description of the size and nature of the incident. For example: how many certificates, OCSP responses, or CRLs were affected; whether the affected objects share features (such as issuance time, signature algorithm, or validation type); and whether the CA Owner had to cease issuance during the incident. 

> If certificates are impacted, the Impact Section MUST include the following information:  

Reviewer: ```suggestion
If certificates are impacted, the Impact section MUST include the following information:  
```
I think we're lowercasing "section" now?

| **Other** | Any other type of "lesson learned" that does not otherwise fit in the above categories, e.g., internal/external circumstances or environmental conditions; discovery of problematic processes, policies, or workflows; communication gaps; resource challenges; task ownership; overlooked warning signs; underutilized tools; etc. Each item mentioned here MUST have at least one corresponding Action Item. |

**Action Items:** The Action Items section contains a list of remediation items that will be undertaken to ensure that similar incidents do not reoccur in the future. For example, if the Whiteboard field contains ‘revocation-delay’, the Action Items list MUST include steps reasonably calculated to prevent or reduce future revocation delays. Note that it is not sufficient for these action items to simply stop this incident, they MUST create additional protections to prevent future incidents.
  
| Field                           | Description                              |  
|---------------------------------|------------------------------------------| 

> | **Action Item Description** | A detailed description of the action to be taken. |

Reviewer: ```suggestion
| **Action Item description** | A detailed description of the action to be taken. |
```

Author: Addressed.

| **Kind** | A classification of whether the action will help Prevent future incidents, Mitigate the impact of future incidents, or Detect future incidents. CA Owners are encouraged to propose action items in all three categories, with an emphasis on Prevent and Mitigate. |
| **Corresponding Root Cause(s)** | The specific Root Cause that the Action intends to remediate (i.e., each problem/issue identified in the "Root Cause Analysis" and "What didn't go well" Sections MUST be mapped to at least one specific action item). |

> | **Evaluation criteria** | Describe how the CA Owner will measure the effectiveness of the Action Item in addressing the Root Cause. Include how the public can also measure this impact, if applicable. CA Owners SHOULD prioritize objective and publicly measurable evidence (i.e., evidence that can be independently verified by the public, such as through Certificate Transparency log data, audit statements, CA policy documents, and public communications (e.g., website content, surveys, etc.)). While external metrics are preferred, these Guidelines recognize that some Action Items may have limited measurable outcomes. When objective metrics are not readily available, CA Owners SHOULD provide a qualitative assessment of the Action Item's effectiveness and explain how they will monitor its impact. Even after an Incident Report has been closed, CA Owners are strongly encouraged to provide periodic updates (e.g., 3, 6, and 12 months post-closure, or at other determined appropriate intervals) related to the ongoing efficacy of the Action Item, as measured against the evaluation criteria. This helps demonstrate a continued commitment to improvement and transparency.

Reviewer: It's **Evaluation criteria** IMO

> | **Evaluation criteria** | Describe how the CA Owner will measure the effectiveness of the Action Item in addressing the Root Cause. Include how the public can also measure this impact, if applicable. CA Owners SHOULD prioritize objective and publicly measurable evidence (i.e., evidence that can be independently verified by the public, such as through Certificate Transparency log data, audit statements, CA policy documents, and public communications (e.g., website content, surveys, etc.)). While external metrics are preferred, these Guidelines recognize that some Action Items may have limited measurable outcomes. When objective metrics are not readily available, CA Owners SHOULD provide a qualitative assessment of the Action Item's effectiveness and explain how they will monitor its impact. Even after an Incident Report has been closed, CA Owners are strongly encouraged to provide periodic updates (e.g., 3, 6, and 12 months post-closure, or at other determined appropriate intervals) related to the ongoing efficacy of the Action Item, as measured against the evaluation criteria. This helps demonstrate a continued commitment to improvement and transparency.

Author: Agree.

> | **Due Date** | A date by which the action item will be complete. |

Reviewer: ```suggestion
| **Due date** | A date by which the action item will be complete. |
```

Author: Addressed.

### Related Incidents

| Bug                                | Date                        | Description                                                            |
|------------------------------------|-----------------------------|------------------------------------------------------------------------|
| [Related Bug ID](Related Bug URL)  | Date Related Bug was opened | A description of how the subject Bug is related to the Bug referenced. |

### Root Cause Analysis

** Issue #: Issue title**
- **Description:** 
- **Issue onset:** 
- **Issue detection:** 
- **Issue resolution:** 
- **Symptoms that led to detection:**
- **How the issue avoided detection:**
- **Root Cause Analysis methodology used:**

### Lessons Learned

- **What went well:** 
- **What didn’t go well:** 
- **Where we got lucky:** 
- **Other:** 

### Action Items

| Action Item | Kind    | Corresponding Root Cause(s) | Evaluation Criteria | Due Date   | Status |
| ----------- | ----    | --------------------------- | ------------------- | -----------| ------ |
| Example     | Prevent | Root Cause # 1              | Criteria            | 2025-01-19 | Value  |

### Appendix

> *Note: see the Incident and Audit Finding Reporting Guidelines on CCADB.org for instructions and expected content.*

Reviewer: Is this note supposed to be here inside of the markdown?

Author: It's location was intentional (otherwise, the template is blank for that section). 

Unbothered if we remove it, though.

Reviewer: No feelings either way.

Author: Eh, killed it. 

Otherwise, it will need to be deleted...every. single. time.

```

#### Closure Report

```markdown

### Report Closure Summary

- **Incident description:** 
- **Incident Root Cause(s):**  
- **Remediation description:**
- **Commitment summary:**  

All Action Items disclosed in this report have been completed as described, and we request its closure.

```

### Report field definitions and expectations

The following sections are intended to describe expected Incident Report and Audit Finding Report content. All fields are mandatory for the corresponding report type, except when described below.

If submitted by the CA Owner corresponding with the report, all fields included in the relevant template MUST be completed. Fields that are not applicable MUST be included in the report and identified as 'N/A'.

#### Incident Reports

**Summary:** The Summary section contains a short description of the nature of the issue and useful background information. This provides just enough context for readers to understand the details in the rest of the report. 

| Field                           | Description                              |  
|---------------------------------|------------------------------------------| 

> | **CA Owner CCADB unique ID*** | The CCADB unique ID value corresponding to the CA Owner's "CA Owner/Certificate" record disclosed in the CCADB. |

Reviewer 2: The CCADB Unique ID begins with the letter "A" followed by a six-digit number and corresponds to the CA Owner record listed in the CCADB.
